test/06-shutdown.test.js

```diff
@@ -0,0 +1,227 @@
+  t.comment(`Copying platform code to ${patchedArtefactDir}`)
+  await fs.promises.mkdir(patchedArtefactDir, { recursive: true })
+  t.teardown(() => Helper.gc(patchedArtefactDir))
+  const mirror = new LocalDrive(rig.artefactDir).mirror(new LocalDrive(patchedArtefactDir), {
+    prune: false,
+    filter: (key) => !(key.startsWith('/pear/') || key.startsWith('/.git/') || key.startsWith('/test/'))
+  })
+  await mirror.done()
+
+  t.comment('Patching sidecar')
+  const sidecarPath = path.join(patchedArtefactDir, 'sidecar.js')
+  const sidecarCode = fs.readFileSync(sidecarPath, 'utf8')
+  const patch = `
+  (() => { require('./constants.js').SPINDOWN_TIMEOUT = ${SPINDOWN_TIMEOUT} })()
+  `
+  const patchedSidecarCode = patch + sidecarCode
+  fs.writeFileSync(sidecarPath, patchedSidecarCode)
+
+  t.comment('Staging patched platform')
+  const rigHelper = new Helper(rig)
+  t.teardown(() => rigHelper.close(), { order: Infinity })
+  await rigHelper.ready()
+
+  const patchedStager = rigHelper.stage({ channel: 'test-spindown-ls', name: 'test-spindown-ls', dir: patchedArtefactDir, dryRun: false, bare: true })
+  const patchedStagerUntil = await Helper.pick(patchedStager, [{ tag: 'final' }])
+  await patchedStagerUntil.final
+
+  t.comment('Seeding patched platform')
+  const patchedSeeder = rigHelper.seed({ channel: 'test-spindown-ls', name: 'test-spindown-ls', dir: patchedArtefactDir, key: null, cmdArgs: [] })
+  const patchedSeederUntil = await Helper.pick(patchedSeeder, [{ tag: 'key' }, { tag: 'announced' }])
+  const patchedPlatformKey = await patchedSeederUntil.key
+  await patchedSeederUntil.announced
+  t.teardown(() => Helper.teardownStream(patchedSeeder))
+
+  t.comment('Bootstrapping patched platform as ls')
+  platformDirLs = path.join(Helper.tmp, 'ls-pear')
+  await Helper.bootstrap(patchedPlatformKey, platformDirLs)
+
+  await Helper.teardownStream(patchedSeeder)
+  await rigHelper.close()
+})
+
+test('sidecar should spindown after a period of inactivity', async (t) => {
+  t.plan(1)
+  t.timeout(SPINDOWN_TIMEOUT + 20_000)
+
+  t.comment('Starting sidecar')
+  const sidecar = spawn(path.join(platformDirLs, 'current', BY_ARCH), ['sidecar'], { stdio: 'pipe' })
+  t.teardown(() => { if (sidecar.exitCode === null) sidecar?.kill() })
+  const untilExit = new Promise(resolve => sidecar.once('exit', resolve))
+  t.teardown(async () => untilExit)
+
+  t.comment(`Waiting for sidecar to spindown (${SPINDOWN_TIMEOUT / 1000}s)`)
+  const timeoutUntil = new Promise((resolve) => {
+    untilExit.finally(() => {
+      if (timeout) clearTimeout(timeout)
+      resolve()
+    })
+    const timeout = setTimeout(() => resolve(false), SPINDOWN_TIMEOUT + 10_000)
+  })
+
+  const hasSpunDown = await Promise.race([untilExit, timeoutUntil])
+  if (hasSpunDown === false) {
+    t.fail('sidecar failed to spin down')
+  } else {
+    t.pass('sidecar has spun down')
+  }
+})
+
+let platformDirRcv
+test.hook('prepare throttled platform', async (t) => {
```

Comment:
Do we need to mark this test as a hook?

Reply:
@rafapaezbas Hmm.. I can merge this in the test where it's being used, but one issue is that the execution time of this step is inconsistent (last run was 30s on Linux and 60s on Windows, but I've seen this step go up to 90s). Right now, I have the timeouts for the spindown test a little tight but if I put this with the test, I'll have to loosen the timeout.

Would it be preferable to put this in the test instead?